src/userobjects/OpenMCDomainFilterEditor.C

```diff
@@ -0,0 +1,206 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCDomainFilterEditor.h"
+#include "OpenMCProblemBase.h"
+#include "openmc/tallies/tally.h"
+#include "openmc/tallies/filter.h"
+#include "openmc/tallies/filter_cell.h"
+#include "openmc/tallies/filter_material.h"
+#include "openmc/tallies/filter_universe.h"
+#include "openmc/tallies/filter_mesh.h"
+
+registerMooseObject("CardinalApp", OpenMCDomainFilterEditor);
+
+InputParameters
+OpenMCDomainFilterEditor::validParams()
+{
+  InputParameters params = OpenMCUserObject::validParams();
+  params.addParam<bool>("create_filter", false, "Whether to create the tally if it doesn't exist");
+  params.addRequiredParam<int32_t>("filter_id", "The ID of the filter to modify");
+  params.addParam<MooseEnum>("filter_type", getFilterTypeEnum(), "The type of filter to create");
+  params.addRequiredParam<std::vector<std::string>>("bins", "The bins to modify in the filter");
+  params.declareControllable("bins");
+  params.addClassDescription("A UserObject for creating and managing OpenMC domain tally filters");
+  return params;
+}
```

Comment:
Very true @nuclearkevin. I wasn't intending these for use with mapped tallies. Unless there's a use case I'm not seeing, I don't think we'd want users controlling those given the amount of programmatic mapping we do in the `OpenMCCellAverageProblem`. For now, I think it's best if we assume these are limited to unmapped tallies.

I could also add some code to preclude their use for mapped Cardinal tallies.

Reply:
I also can't think of any reasons why `OpenMCTallyEditor`/`OpenMCDomainFilterEditor` should be able to modify mapped tallies/filters, so I'm good with restricting their use. It would probably be best to perform a programmatic check when constructing / initializing the userobjects (since they're added after tallies during initialization) to make sure the ids supplied by the users don't conflict with any of the mapped tally/filter ids just to be sure.

If the plan is to avoid the modification of mapped tallies the docs would need to be modified as they [explicitly state](https://github.com/neams-th-coe/cardinal/pull/923/files#diff-fa2f5618663f9bed68907812f88950bd0409609b945497c0c917f25d2f9a7fa2R394) that this is one of the use cases for these userobjects.

Reply:
Agreed! I think that's a good plan. I'll implement a check that ensures the editor IDs and mapped IDs aren't overlapping. I'll look to the documentation too once I've done that. 👍🏻